src/Elastic.Apm.StartupHook.Loader/Loader.cs

```diff
@@ -57,15 +57,24 @@ private static void StartAgent()
 			Agent.Setup(new AgentComponents());
 			Agent.Subscribe(new HttpDiagnosticsSubscriber());
 
-			if (AppDomain.CurrentDomain.GetAssemblies().Any(n => n.GetName().Name.Contains("Microsoft.AspNetCore.")))
+			LoadDiagnosticSubscriber(new AspNetCoreDiagnosticSubscriber());
+			LoadDiagnosticSubscriber(new EfCoreDiagnosticsSubscriber());
+			LoadDiagnosticSubscriber(new SqlClientDiagnosticSubscriber());
+			LoadDiagnosticSubscriber(new ElasticsearchDiagnosticsSubscriber());
+			LoadDiagnosticSubscriber(new GrpcClientDiagnosticSubscriber());
+
+			void LoadDiagnosticSubscriber(IDiagnosticsSubscriber diagnosticsSubscriber)
 			{
-				Agent.Subscribe(
-					new AspNetCoreDiagnosticSubscriber(),
-					new EfCoreDiagnosticsSubscriber(),
-					new SqlClientDiagnosticSubscriber(),
-					new ElasticsearchDiagnosticsSubscriber(),
-					new GrpcClientDiagnosticSubscriber()
-				);
+				var logger = StartupHookLogger.CreateLogger();

```

Comment:
could create once outside the local function and reuse

Reply:
I'm not sure I follow this one. This is the only place currently where we use the logger in `Loader`. Sure, if we'd log later at other places we can move the logger.

Reply:
A new logger is created on every local function instantiation of `LoadDiagnosticSubscriber(...)`, for each DiagnosticSubscriber. Whilst the overhead in creating a new logger each time is small, a single instance could be reused for loading all DiagnosticSubscribers.